### ISSUE TITLE
gillespy2 & pytorch

### ISSUE THREAD

**Issue by ignorant96** [StochSS/GillesPy2#444]
I want to combine gillespy2 with torch to train the parameters of the model, but I failed. I find parameters without changing.So I want to know if gillespy2 can combine with pytorch. Thank You!

**Comment by briandrawert** [comment]
I am not sure about pytorch, but if you are looking to infer model parameter I would strongly suggest Sciope: https://github.com/StochSS/sciope as it is designed specifically for this and works well with GillesPy2.

**Comment by rmjiang7** [comment]
I second the usage of sciope for parameter estimation.   GillesPy2's implementation doesn't use any of PyTorch's data structures so most of PyTorch's features won't work.  In addition, the types of models that GillesPy2 is used for typically have intractable likelihoods/gradients, so the optimization methods in PyTorch won't really be too useful - sciope is built for those type of situations.

**Comment by ignorant96** [comment]
> I second the usage of sciope for parameter estimation. GillesPy2's implementation doesn't use any of PyTorch's data structures so most of PyTorch's features won't work. In addition, the types of models that GillesPy2 is used for typically have intractable likelihoods/gradients, so the optimization methods in PyTorch won't really be too useful - sciope is built for those type of situations.

Thank you so much.I will try sciope.

**Comment by ignorant96** [comment]
> I am not sure about pytorch, but if you are looking to infer model parameter I would strongly suggest Sciope: https://github.com/StochSS/sciope as it is designed specifically for this and works well with GillesPy2.

Thank you so much! I will try it!
